### PR TITLE
ramips-mt7621: add TP-Link Archer C6 v3

### PR DESCRIPTION
**Pull request by Kistelini**
- [x] Must be flashable from vendor firmware
  - [x] Web interface
  - [x] TFTP
  - [ ] Other: <specify>
- [x] Must support upgrade mechanism
  - [x] Must have working sysupgrade
    - [x] Must keep/forget configuration (`sysupgrade [-n]`, `firstboot`)
  - [X] Gluon profile name matches autoupdater image name
        (`lua -e 'print(require("platform_info").get_image_name())'`)
- [x] Reset/WPS/... button must return device into config mode
- [X] Primary MAC address should match address on device label (or packaging)
      (https://gluon.readthedocs.io/en/latest/dev/hardware.html#hardware-support-in-packages)
  - When re-adding a device that was supported by an earlier version of Gluon, a
    factory reset must be performed before checking the primary MAC address, as
    the setting from the old version is not reset otherwise.
- Wired network
  - [X] should support all network ports on the device
  - [X] must have correct port assignment (WAN/LAN)
    - On devices supplied via PoE, there is usually no explicit WAN/LAN labeling on the hardware.
      The PoE input should be the WAN port in this case.
- Wireless network (if applicable)
  - [X] Association with AP must be possible on all radios
  - [X] Association with 802.11s mesh must work on all radios 
  - [X] AP+mesh mode must work in parallel on all radios
- LED mapping
  - Power/system LED
    - [X] Lit while the device is on
    - [x] Should display config mode blink sequence 
          (https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - Radio LEDs
    - [X] Should map to their respective radio
    - [X] Should show activity
  - Switch port LEDs
    - [ ] Should map to their respective port (or switch, if only one led present) 
    - [ ] Should show link state and activity
- Outdoor devices only:
  - [ ] Added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`
- Docs:
  - [X] Added Device to `docs/user/supported_devices.rst`

### CODE REVIEW

**Comment by Kistelini** [comment]
Testing isn't finished. There's a problem with the LAN LED: `gluon-reconfigure` generates this piece for `/etc/config/system`
```
config led 'led_lan'
	option name 'LAN'
	option sysfs 'green:lan'
	option trigger 'netdev'
	option mode 'link tx rx'
	option dev 'br-lan'
```
because of [these defaults](https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds#L150), but Gluon doesn't have `br-lan`  so the LED doesn't work. How do I fix this?

**Comment by Kistelini** [comment]
Those `mt7530-*` triggers don't do anything. Everything else works!

**Comment by blocktrron** [comment]
I think we have to take a look at software-controlled LEDs spanning multiple LAN ports in conjunction with DSA, as this is currently not handled nicely upstream. I assume some other devices have lost sane handling with migrating to DSA also.

File: targets/ramips-mt7621 [hunk 1]
```diff
@@ -49,6 +49,10 @@ device('netgear-wndr3700-v5', 'netgear_wndr3700-v5', {
 
 -- TP-Link
 
+device('tp-link-archer-c6-v3', 'tplink_archer-c6-v3', {
+	broken = true, -- LAN LED not working
```
Comment (AiyionPrime) [inline targets/ramips-mt7621:53]:
```suggestion
	broken = true, -- LAN LED not working - review after resolving #2756
```

**Review by AiyionPrime — CHANGES_REQUESTED** [review]
@Kistelini Sorry this took so long, took a moment to understand how swconfig did it and why this is not trivial to fix upstream.
I'd like to add the expiration note for when the broken status is expected to be fixed, but other than that the PR looks fine.

**Comment by Djfe** [comment]
Should we add the Archer A6 v3 as well? They share the exact same dtsi/dts, the same led config etc. They just have a different name, which might influence the tp-link safeloader. (so it probably has to be a different entry instead of an alias, right?)

**Comment by AiyionPrime** [comment]
As soon as someone wants to use the device, we've got someone to test it.
We usually do not merge untested devices anymore. Even if we think the difference is minor and it should just work.
Whoever wants to use and test it, will have a smooth run integrating it :)

**Comment by Djfe** [comment]
@AiyionPrime backport needed

**Comment by AiyionPrime** [comment]
> @AiyionPrime backport needed

Won't happen, it's broken.

We'll publish the list of upcoming backports in the tracking issue tomorrow morning again. (Hoping GitHub does not eat it twice -.-')

**Pull request merged**